src/gpuarray_array.c

```diff
@@ -440,27 +457,35 @@ int GpuArray_take1(GpuArray *a, const GpuArray *v, const GpuArray *i,
   int addr32 = 0;
 
   if (!GpuArray_ISWRITEABLE(a))
-    return GA_INVALID_ERROR;
+    return error_set(ctx->err, GA_INVALID_ERROR, "Destination array (a) not writeable");
 
   if (!GpuArray_ISALIGNED(a) || !GpuArray_ISALIGNED(v) ||
       !GpuArray_ISALIGNED(i))
-    return GA_UNALIGNED_ERROR;
+    return error_fmt(ctx->err, GA_UNALIGNED_ERROR,
+                     "Not all arrays are aligned: a (%d), b (%d), i (%d)",
+                     GpuArray_ISALIGNED(a), GpuArray_ISALIGNED(v), GpuArray_ISALIGNED(i));
 
   /* a and i have to be C contiguous */
-  if (!GpuArray_IS_C_CONTIGUOUS(a) || !GpuArray_IS_C_CONTIGUOUS(i))
-    return GA_INVALID_ERROR;
+  if (!GpuArray_IS_C_CONTIGUOUS(a))
+    return error_set(ctx->err, GA_INVALID_ERROR, "Destination array (a) not C-contiguous");
+  if (!GpuArray_IS_C_CONTIGUOUS(i))
+    return error_set(ctx->err, GA_INVALID_ERROR, "Index array (i) not C-contiguous");
 
   /* Check that the dimensions match namely a[0] == i[0] and a[>0] == v[>0] */
   if (v->nd == 0 || a->nd == 0 || i->nd != 1 || a->nd != v->nd ||
       a->dimensions[0] != i->dimensions[0])
-    return GA_INVALID_ERROR;
+    return error_fmt(ctx->err, GA_INVALID_ERROR, "Dimension mismatch. "
+                     "v->nd = %llu, a->nd = %llu, i->nd = %llu, a->dimensions[0] = %llu, i->dimensions[0] = %llu",
+                     v->nd, a->nd, i->nd, a->nd > 0 ? a->dimensions[0] : 0, i->nd > 1 ? i->dimensions[0] : 1);
```

Comment:
Should have been `i->nd > 0` indeed.
But I'm not sure what to print in the case where `nd == 0`. Does `dimensions` even exist? I wanted to print 1 because there is only one element in total and it would have fitted `%llu`, but maybe we need to rework how the errors are detected in two phases.